doc/source/user_guide/tutorials/mesh/create_a_mesh_from_scratch.rst

```diff
@@ -0,0 +1,158 @@
+                if np.allclose(arr[index + 1], seq[1]) and np.allclose(arr[index + 2], seq[2]):
+                    return index
+        return -1
+
+Add nodes
+---------
+
+Add |Nodes| to the |MeshedRegion| object.
+
+.. jupyter-execute::
+
+    node_id = 1
+    for i, x in enumerate(
+        [float(i) * length / float(num_nodes_in_length) for i in range(0, num_nodes_in_length)]
+    ):
+        for j, y in enumerate(
+            [float(i) * width / float(num_nodes_in_width) for i in range(0, num_nodes_in_width)]
+        ):
+            for k, z in enumerate(
+                [float(i) * depth / float(num_nodes_in_depth) for i in range(0, num_nodes_in_depth)]
+            ):
+                my_meshed_region.nodes.add_node(node_id, [x, y, z])
+                node_id += 1
+
+Get the nodes coordinates field.
+
+.. jupyter-execute::
+
+    my_nodes_coordinates = my_meshed_region.nodes.coordinates_field
+
+Set the mesh properties
+-----------------------
+
+Set the mesh unit.
+
+.. jupyter-execute::
+
+    my_meshed_region.unit = "mm"
+
+Set the nodes coordinates.
+
+.. jupyter-execute::
+
+    # Get the nodes coordinates data
+    my_nodes_coordinates_data = my_nodes_coordinates.data
+    # As we use the connectivity function we need to get the data as a list
+    my_nodes_coordinates_data_list = my_nodes_coordinates.data_as_list
+    # Set the nodes scoping
+    my_coordinates_scoping = my_nodes_coordinates.scoping
+
+Add elements
+------------
+Add |Elements| to the |MeshedRegion| object.
+
+.. jupyter-execute::
+
+    element_id = 1
+    for i, x in enumerate(
+        [float(i) * length / float(num_nodes_in_length) for i in range(num_nodes_in_length - 1)]
+    ):
+        for j, y in enumerate(
+            [float(i) * width / float(num_nodes_in_width) for i in range(num_nodes_in_width - 1)]
+        ):
+            for k, z in enumerate(
+                [float(i) * depth / float(num_nodes_in_depth) for i in range(num_nodes_in_depth - 1)]
+            ):
+                coord1 = np.array([x, y, z])
+                connectivity = []
+                for xx in [x, x + length / float(num_nodes_in_length)]:
+                    for yy in [y, y + width / float(num_nodes_in_width)]:
+                        for zz in [z, z + depth / float(num_nodes_in_depth)]:
+                            data_index = search_sequence_numpy(my_nodes_coordinates_data_list, [xx, yy, zz])
+                            scoping_index = int(data_index / 3)  # 3components
+                            connectivity.append(scoping_index)
+                # rearrange connectivity
+                tmp = connectivity[2]
+                connectivity[2] = connectivity[3]
+                connectivity[3] = tmp
+                tmp = connectivity[6]
+                connectivity[6] = connectivity[7]
+                connectivity[7] = tmp
+                my_meshed_region.elements.add_solid_element(element_id, connectivity)
+                element_id += 1
```

Comment:
By storing constants in variables, the readability of this code can be improved:

```suggestion
    element_id = 1

    # Precompute node spacings
    dx = length / float(num_nodes_in_length)
    dy = width / float(num_nodes_in_width)
    dz = depth / float(num_nodes_in_depth)

    # Generate node coordinates
    x_coords = [i * dx for i in range(num_nodes_in_length - 1)]
    y_coords = [j * dy for j in range(num_nodes_in_width - 1)]
    z_coords = [k * dz for k in range(num_nodes_in_depth - 1)]

    # Iterate through the grid
    for x in x_coords:
        for y in y_coords:
            for z in z_coords:
                coord1 = np.array([x, y, z])
                connectivity = []

                # Generate connectivity for the current element
                for xx in [x, x + dx]:
                    for yy in [y, y + dy]:
                        for zz in [z, z + dz]:
                            data_index = search_sequence_numpy(my_nodes_coordinates_data_list, [xx, yy, zz])
                            scoping_index = int(data_index / 3)  # 3 components per node
                            connectivity.append(scoping_index)

                # Rearrange connectivity to maintain element orientation
                connectivity[2], connectivity[3] = connectivity[3], connectivity[2]
                connectivity[6], connectivity[7] = connectivity[7], connectivity[6]

                # Add the solid element
                my_meshed_region.elements.add_solid_element(element_id, connectivity)
                element_id += 1
```